mirai-core-api/src/commonMain/kotlin/message/data/SuperFace.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019-2023 Mamoe Technologies and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
+ *
+ * https://github.com/mamoe/mirai/blob/dev/LICENSE
+ */
+
+package net.mamoe.mirai.message.data
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import net.mamoe.mirai.message.code.CodableMessage
+import net.mamoe.mirai.message.data.visitor.MessageVisitor
+import net.mamoe.mirai.utils.MiraiExperimentalApi
+import net.mamoe.mirai.utils.MiraiInternalApi
+import net.mamoe.mirai.utils.NotStableForInheritance
+import net.mamoe.mirai.utils.safeCast
+import kotlin.jvm.Throws
+
+/**
+ * 超级表情
+ *
+ * @see Face
+ */
+@OptIn(MiraiExperimentalApi::class)
+@Serializable
+@SerialName(SuperFace.SERIAL_NAME)
+@NotStableForInheritance
+public data class SuperFace @MiraiInternalApi constructor(
```

Comment:
No data class
https://kotlinlang.org/docs/jvm-api-guidelines-backward-compatibility.html#don-t-use-data-classes-in-an-api